src/promotion/promotion.c

```diff
@@ -83,6 +106,23 @@ ocf_error_t ocf_promotion_set_policy(ocf_promotion_policy_t policy,
 		return 0;
 	}
 
+	if (ocf_promotion_policies[type].size_of) {
+		size = ocf_promotion_policies[type].size_of(cache);
+		available = env_get_free_memory();
+
+		if (size >= available) {
+			ocf_cache_log(cache, log_err, "Not enough memory to "
+					"initialize '%s' promotion policy! "
+					"Required %lu, available %lu. "
+					"Policy didn't change\n",
+					ocf_promotion_policies[type].name,
+					(long unsigned)size,
+					(long unsigned)available);
+
+			return -OCF_ERR_NO_MEM;
```

Comment:
Is it enough for casadm to know exactly what message it should print? I guess `-OCF_ERR_NO_MEM` could happen also in other places (e.g. cache management lock can also return `-OCF_ERR_NO_MEM`). In case of cache start, when we don't have enough available memory we return `-OCF_ERR_NO_FREE_RAM` and it also seems to be better fit here.

Reply:
True that, changed